### ISSUE TITLE
Error spam when running the test suite

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#80616]
### Godot version

4.2.dev (

### System information

Mageia 9 - Vulkan (Forward+) - dedicated AMD Radeon RX Vega M GL Graphics (RADV VEGAM) () - Intel(R) Core(TM) i7-8705G CPU @ 3.10GHz (8 Threads)

### Issue description

When compiling Godot with `dev_build=yes tests=yes`, running the test suite succeeds, but prints a lot of errors:
```
$ ./bin/godot.linuxbsd.editor.dev.x86_64 --test
[doctest] doctest version is "2.4.11"
[doctest] run with "--help" for options
ERROR: Condition "singleton != nullptr" is true.
   at: NavigationServer3D (./servers/navigation_server_3d.cpp:189)
ERROR: Resource was not pre cached for the resource section, bug?
   at: _write_resource (./scene/resources/resource_format_text.cpp:1848)
WARNING: Couldn't load resource (no cache): local://Resource_q0hwh
     at: parse_variant (./core/io/resource_format_binary.cpp:411)
ERROR: The object does not have any 'meta' values with the key 'next'.
   at: get_meta (./core/object/object.cpp:926)
ERROR: The object does not have any 'meta' values with the key 'next'.
   at: get_meta (./core/object/object.cpp:926)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point (./modules/navigation/godot_navigation_server.cpp:241)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point_normal (./modules/navigation/godot_navigation_server.cpp:248)
ERROR: Condition "map == nullptr" is true. Returning: RID()
   at: map_get_closest_point_owner (./modules/navigation/godot_navigation_server.cpp:255)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point_to_segment (./modules/navigation/godot_navigation_server.cpp:234)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point_to_segment (./modules/navigation/godot_navigation_server.cpp:234)
ERROR: Condition "map == nullptr" is true. Returning: Vector<Vector3>()
   at: map_get_path (./modules/navigation/godot_navigation_server.cpp:227)
ERROR: Condition "map == nullptr" is true. Returning: Vector<Vector3>()
   at: map_get_path (./modules/navigation/godot_navigation_server.cpp:227)
ERROR: Condition "map == nullptr" is true. Returning: r_query_result
   at: _query_path (./modules/navigation/godot_navigation_server.cpp:1161)
ERROR: Condition "!map" is true. Returning: Vector3()
   at: get_connection_pathway_end (./modules/navigation/nav_region.cpp:98)
ERROR: Condition "!map" is true. Returning: Vector3()
   at: get_connection_pathway_start (./modules/navigation/nav_region.cpp:92)
WARNING: NavigationServer3D::region_bake_navigation_mesh() is deprecated due to core threading changes. To upgrade existing code, first create a NavigationMeshSourceGeometryData3D resource. Use this resource with method parse_source_geometry_data() to parse the SceneTree for nodes that should contribute to the navigation mesh baking. The SceneTree parsing needs to happen on the main thread. After the parsing is finished use the resource with method bake_from_source_geometry_data() to bake a navigation mesh..
     at: region_bake_navigation_mesh (./modules/navigation/godot_navigation_server.cpp:468)
===============================================================================
[doctest] test cases:     816 |     816 passed | 0 failed | 1 skipped
[doctest] assertions: 2331757 | 2331757 passed | 0 failed |
[doctest] Status: SUCCESS!
WARNING: ObjectDB instances leaked at exit (run with --verbose for details).
     at: cleanup (./core/object/object.cpp:2049)
```

Some of those errors are intentionally caused by the tests, but they need to be silenced with `ERR_PRINT_OFF` and `ERR_PRINT_ON` macros in the test.

CC @smix8 as most are navigation related.

### Steps to reproduce

- `scons dev_build=yes tests=yes`
- `godot --test`

### Minimal reproduction project

n/a

**Comment by akien-mga** [comment]
CC @godotengine/tests

**Comment by smix8** [comment]
Those navigation related errors here will trigger when the functions are called without a navigation map sync. If those errors are unwanted basically all those newer navigation related tests (CC @Scony) need to be wrapped with `ERR_PRINT_OFF` / `ERR_PRINT_ON` to suppress them.

**Comment by Scony** [comment]
If this is possible to disable those logs, then it's good idea - I'll experiment with that

**Comment by akien-mga** [comment]
The navigation errors were fixed by #80833, so as of 8c1817f75 this is the output:
```
ERROR: Resource was not pre cached for the resource section, bug?
   at: _write_resource (./scene/resources/resource_format_text.cpp:1848)
WARNING: Couldn't load resource (no cache): local://Resource_q0hwh
     at: parse_variant (./core/io/resource_format_binary.cpp:411)
ERROR: The object does not have any 'meta' values with the key 'next'.
   at: get_meta (./core/object/object.cpp:926)
ERROR: The object does not have any 'meta' values with the key 'next'.
   at: get_meta (./core/object/object.cpp:926)
```

CC @KoBeWi

**Comment by AThousandShips** [comment]
I've identified one of the parts, will open a PR for it
